test/common/upstream/zone_aware_load_balancer_fuzz_base.cc

```diff
@@ -0,0 +1,70 @@
+#include "zone_aware_load_balancer_fuzz_base.h"
+
+#include "test/mocks/upstream/host_set.h"
+
+namespace Envoy {
+namespace Upstream {
+
+void ZoneAwareLoadBalancerFuzzBase::initializeASingleHostSet(
+    const test::common::upstream::SetupPriorityLevel& setup_priority_level,
+    const uint8_t priority_level, uint16_t& port) {
+  LoadBalancerFuzzBase::initializeASingleHostSet(setup_priority_level, priority_level, port);
+  // Update local priority set if it exists - will mean load balancer is zone aware and has decided
+  // to construct local priority set
+  if (priority_level == 0 && local_priority_set_.get() != nullptr) {
+    MockHostSet& host_set = *priority_set_.getMockHostSet(priority_level);
+    local_priority_set_->getMockHostSet(0)->hosts_ = host_set.hosts_;
+    local_priority_set_->getMockHostSet(0)->hosts_per_locality_ = host_set.hosts_per_locality_;
+  }
+}
+
+void ZoneAwareLoadBalancerFuzzBase::updateHealthFlagsForAHostSet(
+    const uint64_t host_priority, const uint32_t num_healthy_hosts,
+    const uint32_t num_degraded_hosts, const uint32_t num_excluded_hosts,
+    const std::string random_bytestring) {
+  LoadBalancerFuzzBase::updateHealthFlagsForAHostSet(
+      host_priority, num_healthy_hosts, num_degraded_hosts, num_excluded_hosts, random_bytestring);
+  // Update local priority set if it exists - will mean load balancer is zone aware and has decided
+  // to construct local priority set
+  const uint8_t priority_of_host_set = host_priority % num_priority_levels_;
+  if (priority_of_host_set == 0 && local_priority_set_.get() != nullptr) {
```

Comment:
to test unique ptr existence, you can just use it's bool operator
http://www.cplusplus.com/reference/memory/unique_ptr/operator%20bool/

Reply:
Wow, good to know, thanks!

Reply:
Deleted.